Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2015-11-16
Initial Package Version: 2.7.8
Upstream Status: Applied
Origin: Upstream
Description: Fixes build failure if libgsl >= 2.0 is used.

diff -Naur a/gsl.cc b/gsl.cc
--- a/gsl.cc	2015-05-19 23:23:54.000000000 +0100
+++ b/gsl.cc	2015-11-16 20:20:18.819563505 +0000
@@ -27,6 +27,7 @@
 #include <gsl/gsl_rng.h>
 #include <gsl/gsl_randist.h>
 #include <gsl/gsl_cdf.h>
+#include <gsl/gsl_version.h>
 
 #include "opsymbols.h"
 
@@ -1088,7 +1089,11 @@
   addGSLDOUBLE2Func<gsl_sf_ellint_F>(SYM(F));
   addGSLDOUBLE2Func<gsl_sf_ellint_E>(SYM(E));
   addGSLDOUBLE3Func<gsl_sf_ellint_P>(SYM(P),SYM(phi),SYM(k),SYM(n));
+#if GSL_MAJOR_VERSION >= 2
+  addGSLDOUBLE2Func<gsl_sf_ellint_D>(SYM(D),SYM(phi),SYM(k));
+#else
   addGSLDOUBLE3Func<gsl_sf_ellint_D>(SYM(D),SYM(phi),SYM(k),SYM(n));
+#endif
   addGSLDOUBLE2Func<gsl_sf_ellint_RC>(SYM(RC),SYM(x),SYM(y));
   addGSLDOUBLE3Func<gsl_sf_ellint_RD>(SYM(RD),SYM(x),SYM(y),SYM(z));
   addGSLDOUBLE3Func<gsl_sf_ellint_RF>(SYM(RF),SYM(x),SYM(y),SYM(z));
